tests/utilities/test_fetching.py

```diff
@@ -0,0 +1,70 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import pytest
+from torch import tensor
+from torch.utils.data import DataLoader, IterableDataset
+
+from pytorch_lightning.trainer.supporters import CombinedLoader
+from pytorch_lightning.utilities.fetching import LightningFetcher
+
+
+@pytest.mark.parametrize("use_combined_loader", [False, True])
+def test_prefetch_iterator(use_combined_loader):
```

Comment:
How will the LightningFetcher be sent to the Trainer? From offline discussions I recall it being like so:

```python
trainer = Trainer(lightning_fetcher=CustomLightningFetcher())
```
is that so? If so cool :)

Reply:
Yes, something like that in the future :)

Reply:
trainer = Trainer(fetched_batches=[0, 1, 2, "auto", CustomLightningFetcher(batch_processor=...)])